### PR TITLE
[14.0] [MIG] report_qr 

### PR DESCRIPTION
**Pull request by Kiplangatdan**

### CODE REVIEW

**Comment by gjotten** [comment]
@thomaspaulb for you to take a look at

**Review by etobella — APPROVED** [review]
Code Review
LGTM

Waiting for runbot in order to test it

**Review by thomaspaulb — APPROVED** [review]
Checked against [this](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0) and standard migration looks fine to me.

**Review by etobella — APPROVED** [review]
Tested with runbot

File: report_qr/README.rst [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+Authors
+~~~~~~~
+
+* Creu Blanca
+
+Contributors
+~~~~~~~~~~~~
+
+* Enric Tobella <etobella@creublanca.es>
```
Comment (gfcapalbo) [inline report_qr/README.rst:90]: @Kiplangatdan add yourself as a contributor

File: report_qr/models/ir_actions_report.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import io
+
+import qrcode
```
Comment (gfcapalbo) [inline report_qr/models/ir_actions_report.py:3]: i think qrcode should be added as a external library dependency , and perhaps in the repo requirements. https://github.com/OCA/reporting-engine/blob/14.0/requirements.txt
unless there isn't some indirect dependency that guarantees qrcode to be installed?

File: report_qr/tests/test_report_qr.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import HttpCase
+
+
+class TestReportQr(HttpCase):
+    def test_qr_generation(self):
+        data = "TEST"
+        image = self.url_open("/report/qr?value=%s" % data)
+        self.assertEqual(image.headers["Content-type"], "image/png")
```
Comment (gfcapalbo) [inline report_qr/tests/test_report_qr.py:11]: maybe also test for correct encoding by decoding  the image back to text?
self.assertEqual( qrcode.Decoder().decode(image).result , "TEST")

**Review by gfcapalbo — APPROVED** [review]
some minor ideas about dependencies and testing, nothing substantial, looks good to merge

**Pull request merged**